src/coupled_driver.cpp

```diff
@@ -401,18 +408,23 @@ void CoupledDriver::init_densities()
 {
   comm_.message("Initializing densities");
 
-  if (this->has_global_coupling_data()) {
-    auto n_global = this->get_heat_driver().n_global_elem();
+  const auto& neutronics = this->get_neutronics_driver();
+  const auto& heat = this->get_heat_driver();
+
+  auto n_global = this->get_heat_driver().n_global_elem();
+
+  if (neutronics.comm_.is_root()) {
```

Comment:
Again, this seems wrong to me. Wouldn't the value of `n_global` be invalid on the neutronics comm if it is disjoint from the heat comm?